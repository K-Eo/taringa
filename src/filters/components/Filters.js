import React from 'react'
import { NavLink } from 'react-router-dom'
import classNames from 'classnames'

const Filters = ({ className, dispatch, items, ...rest }) => {
  const classes = classNames(className, 'navbar-nav mr-auto')

  return (
    <div className={classes} {...rest}>
      {items.map(filter => (
        <NavLink
          activeClassName="active"
          className="nav-item nav-link"
          exact={filter.exact}
          key={filter.id}
          to={{
            pathname: filter.pathname,
            state: { clear: true },
          }}
        >
          {filter.displayName}
        </NavLink>
      ))}
    </div>
  )
}

Filters.defaultProps = {
  items: [],
}

export default Filters
