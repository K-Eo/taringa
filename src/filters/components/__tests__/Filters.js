import React from 'react'

import Filters from '../Filters'

describe('Filters', () => {
  it('exists', () => {
    expect(Filters).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<Filters />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with additional classes', () => {
    const wrapper = shallow(<Filters className="foobar" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with additional props', () => {
    const wrapper = shallow(<Filters id="foobar" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders filters', () => {
    const filters = [
      {
        displayName: 'Foo',
        exact: true,
        id: 'foo',
        pathname: '/',
      },
      {
        displayName: 'Bar',
        exact: false,
        id: 'bar',
        pathname: '/bar',
      },
    ]

    const wrapper = shallow(<Filters items={filters} />)

    expect(wrapper).toMatchSnapshot()
  })
})
