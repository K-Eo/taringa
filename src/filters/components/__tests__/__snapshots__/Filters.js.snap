// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters renders 1`] = `
<div
  className="navbar-nav mr-auto"
/>
`;

exports[`Filters renders filters 1`] = `
<div
  className="navbar-nav mr-auto"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="nav-item nav-link"
    exact={true}
    key="foo"
    to={
      Object {
        "pathname": "/",
        "state": Object {
          "clear": true,
        },
      }
    }
  >
    Foo
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="nav-item nav-link"
    exact={false}
    key="bar"
    to={
      Object {
        "pathname": "/bar",
        "state": Object {
          "clear": true,
        },
      }
    }
  >
    Bar
  </NavLink>
</div>
`;

exports[`Filters renders with additional classes 1`] = `
<div
  className="foobar navbar-nav mr-auto"
/>
`;

exports[`Filters renders with additional props 1`] = `
<div
  className="navbar-nav mr-auto"
  id="foobar"
/>
`;
